File: src/main/java/com/eager/questioncloud/domain/subscribe/repository/SubscribeJpaRepository.java
package com.eager.questioncloud.domain.subscribe.repository;

import com.eager.questioncloud.domain.subscribe.entity.SubscribeEntity;
import org.springframework.data.jpa.repository.JpaRepository;

public interface SubscribeJpaRepository extends JpaRepository<SubscribeEntity, Long> {
    Boolean existsBySubscriberIdAndCreatorId(Long subscriberId, Long creatorId);

    void deleteBySubscriberIdAndCreatorId(Long subscriberId, Long creatorId);

    int countByCreatorId(Long creatorId);
}

File: qc-core/qc-subscribe/qc-subscribe-rdb/src/main/kotlin/com/eager/questioncloud/subscribe/repository/SubscribeJpaRepository.kt
package com.eager.questioncloud.subscribe.repository

import com.eager.questioncloud.subscribe.entity.SubscribeEntity
import org.springframework.data.jpa.repository.JpaRepository

interface SubscribeJpaRepository : JpaRepository<SubscribeEntity, Long> {
    fun existsBySubscriberIdAndCreatorId(subscriberId: Long, creatorId: Long): Boolean
    
    fun deleteBySubscriberIdAndCreatorId(subscriberId: Long, creatorId: Long)
    
    fun countByCreatorId(creatorId: Long): Int
}
